applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -0,0 +1,588 @@
+    PropertiesType& prop = this->GetProperties();
+
+    //GeometryData::IntegrationMethod rIntegrationMethod = GeometryData::IntegrationMethod::GI_GAUSS_1;
+    GeometryData::IntegrationMethod rIntegrationMethod = this->mThisIntegrationMethod;
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = Geom.IntegrationPoints(rIntegrationMethod);
+    const unsigned int NumGPoints = IntegrationPoints.size();
+    const unsigned int LocalDim = Geom.LocalSpaceDimension();
+
+    //Containers of variables at all integration points
+    const Matrix& NContainer = Geom.ShapeFunctionsValues(rIntegrationMethod);
+    GeometryType::JacobiansType JContainer(NumGPoints);
+    for (unsigned int i = 0; i < NumGPoints; ++i)
+        (JContainer[i]).resize(TDim, LocalDim, false);
+    Geom.Jacobian(JContainer, rIntegrationMethod);
+
+    //Condition variables
+    BoundedMatrix<double, TDim, TNumNodes* TDim> Nu = ZeroMatrix(TDim, TNumNodes * TDim);
+
+    NormalLysmerAbsorbingVariables rVariables;
+
+    this->GetVariables(rVariables, CurrentProcessInfo);
+    this->CalculateNodalStiffnessMatrix(rVariables, CurrentProcessInfo, Geom);
+
+    BoundedMatrix<double, TDim, TNumNodes* TDim> AuxAbsKMatrix;
+    BoundedMatrix<double, TNumNodes* TDim, TNumNodes* TDim> rAbsKMatrix = ZeroMatrix(n_dof, n_dof);
+
+    //Loop over integration points
+    for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+
+        // calculate displacement shape function matrix
+        GeoElementUtilities::CalculateNuMatrix<TDim, TNumNodes>(Nu, NContainer, GPoint);
+
+        //Compute weighting coefficient for integration
+        this->CalculateIntegrationCoefficient(rVariables.IntegrationCoefficient,
+            JContainer[GPoint],
+            IntegrationPoints[GPoint].Weight());
+
+
+        // set stiffness part of absorbing matrix
+        AuxAbsKMatrix = prod(rVariables.KAbsMatrix, Nu);
+        rAbsKMatrix += prod(trans(Nu), AuxAbsKMatrix) * rVariables.IntegrationCoefficient;
+
+    }
+    rVariables.UMatrix = rAbsKMatrix;
+
+    // assemble left hand side vector
+    const SizeType ConditionSize = TNumNodes * TDim + TNumNodes;
+    noalias(rLhsMatrix) = ZeroMatrix(ConditionSize);
+    if (rLhsMatrix.size1() != ConditionSize)
+        rLhsMatrix.resize(ConditionSize, ConditionSize, false);
+
+    noalias(rLhsMatrix) = ZeroMatrix(ConditionSize, ConditionSize);
+    this->CalculateAndAddLHS(rLhsMatrix, rVariables);
+
+    // no righthand side contribution
+    rRightHandSideVector = ZeroVector(ConditionSize);
+}
+
+/// <summary>
+/// Calculates LHS Damping part of abosrbing boundary
+/// </summary>
+/// <param name="rDampingMatrix"></param>
+/// <param name="CurrentProcessInfo"></param>
+template< unsigned int TDim, unsigned int TNumNodes >
+void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& CurrentProcessInfo)
+{
+    int n_dof = TNumNodes * TDim;
```

Comment:
can be removed if it is defined as

static constexpr SizeType N_DOF = TNumNodes * TDim;

in the header file.